cognitive/src/main/scala/com/microsoft/azure/synapse/ml/cognitive/CognitiveServiceBase.scala

```diff
@@ -132,14 +132,78 @@ trait HasSubscriptionKey extends HasServiceParams {
 
   def getSubscriptionKey: String = getScalarParam(subscriptionKey)
 
-  def setSubscriptionKey(v: String): this.type = setScalarParam(subscriptionKey, v)
+  def setSubscriptionKey(v: String): this.type = {
+    println("WARNING: Please use setSubscriptionKey together with setLocation for authentication")
+    setScalarParam(subscriptionKey, v)
+  }
 
   def getSubscriptionKeyCol: String = getVectorParam(subscriptionKey)
 
   def setSubscriptionKeyCol(v: String): this.type = setVectorParam(subscriptionKey, v)
 
 }
 
+trait HasAADToken extends HasServiceParams {
+  val aadToken = new ServiceParam[String](
+    this, "aadToken", "AAD Token used for authentication"
+  )
+
+  def setAadToken(v: String): this.type = {
+    println("WARNING: Please use setAadToken together with setEndpoint/setServiceName for authentication")
+    setScalarParam(aadToken, v)
+  }
+
+  def getAadToken: String = getScalarParam(aadToken)
+
+  def setAadTokenCol(v: String): this.type = setVectorParam(aadToken, v)
+
+  def getAadTokenCol: String = getVectorParam(aadToken)
+}
+
+trait HasCustomDomain extends Wrappable with HasURL with HasUrlPath {
```

Comment:
HasCustomCogServiceDomain?  HasCustomDomain seems a little generic for what it does.